### ISSUE TITLE
libunwind.dylib', file was built for x86_64 which is not the architecture being linked (i386)

### ISSUE THREAD

**Issue by paddor** [pbhogan/scrypt#64]
My system is OSX (64 bit) and I get this error when trying to install scrypt:

```
$ gem install scrypt -v '3.0.3'
Building native extensions.  This could take a while...

ERROR:  Error installing scrypt:
    ERROR: Failed to build gem native extension.

    current directory: /Users/paddor/.gem/ruby/2.3.1/gems/scrypt-3.0.3/ext/scrypt
/Users/paddor/.rubies/ruby-2.3.1/bin/ruby -rubygems /Users/paddor/.gem/ruby/2.3.1/gems/rake-11.2.2/exe/rake RUBYARCHDIR=/Users/paddor/.gem/ruby/2.3.1/extensions/x86_64-darwin-15/2.3.0-static/scrypt-3.0.3 RUBYLIBDIR=/Users/paddor/.gem/ruby/2.3.1/extensions/x86_64-darwin-15/2.3.0-static/scrypt-3.0.3
mkdir -p x86_64-darwin
clang -fexceptions -O -fno-omit-frame-pointer -fno-strict-aliasing -Wall -std=c99 -msse -msse2 -arch x86_64 -arch i386   -o x86_64-darwin/crypto_scrypt.o -c ./crypto_scrypt.c
clang -fexceptions -O -fno-omit-frame-pointer -fno-strict-aliasing -Wall -std=c99 -msse -msse2 -arch x86_64 -arch i386   -o x86_64-darwin/crypto_scrypt_smix.o -c ./crypto_scrypt_smix.c
clang -fexceptions -O -fno-omit-frame-pointer -fno-strict-aliasing -Wall -std=c99 -msse -msse2 -arch x86_64 -arch i386   -o x86_64-darwin/crypto_scrypt_smix_sse2.o -c ./crypto_scrypt_smix_sse2.c
clang -fexceptions -O -fno-omit-frame-pointer -fno-strict-aliasing -Wall -std=c99 -msse -msse2 -arch x86_64 -arch i386   -o x86_64-darwin/insecure_memzero.o -c ./insecure_memzero.c
clang -fexceptions -O -fno-omit-frame-pointer -fno-strict-aliasing -Wall -std=c99 -msse -msse2 -arch x86_64 -arch i386   -o x86_64-darwin/memlimit.o -c ./memlimit.c
clang -fexceptions -O -fno-omit-frame-pointer -fno-strict-aliasing -Wall -std=c99 -msse -msse2 -arch x86_64 -arch i386   -o x86_64-darwin/scrypt_calibrate.o -c ./scrypt_calibrate.c
clang -fexceptions -O -fno-omit-frame-pointer -fno-strict-aliasing -Wall -std=c99 -msse -msse2 -arch x86_64 -arch i386   -o x86_64-darwin/scrypt_ext.o -c ./scrypt_ext.c
clang -fexceptions -O -fno-omit-frame-pointer -fno-strict-aliasing -Wall -std=c99 -msse -msse2 -arch x86_64 -arch i386   -o x86_64-darwin/scryptenc_cpuperf.o -c ./scryptenc_cpuperf.c
clang -fexceptions -O -fno-omit-frame-pointer -fno-strict-aliasing -Wall -std=c99 -msse -msse2 -arch x86_64 -arch i386   -o x86_64-darwin/sha256.o -c ./sha256.c
clang -fexceptions -O -fno-omit-frame-pointer -fno-strict-aliasing -Wall -std=c99 -msse -msse2 -arch x86_64 -arch i386   -o x86_64-darwin/warnp.o -c ./warnp.c
clang -bundle -o x86_64-darwin/libscrypt_ext.bundle x86_64-darwin/crypto_scrypt.o x86_64-darwin/crypto_scrypt_smix.o x86_64-darwin/crypto_scrypt_smix_sse2.o x86_64-darwin/insecure_memzero.o x86_64-darwin/memlimit.o x86_64-darwin/scrypt_calibrate.o x86_64-darwin/scrypt_ext.o x86_64-darwin/scryptenc_cpuperf.o x86_64-darwin/sha256.o x86_64-darwin/warnp.o -fexceptions -arch x86_64 -arch i386
ld: in '/usr/local/lib/libunwind.dylib', file was built for x86_64 which is not the architecture being linked (i386): /usr/local/lib/libunwind.dylib for architecture i386
clang-3.8: error: linker command failed with exit code 1 (use -v to see invocation)
rake aborted!
Command failed with status (1): [clang -bundle -o x86_64-darwin/libscrypt_e...]
/Users/paddor/.gem/ruby/2.3.1/gems/ffi-compiler-1.0.1/lib/ffi-compiler/compile_task.rb:153:in `block in define_task!'
/Users/paddor/.gem/ruby/2.3.1/gems/rake-11.2.2/exe/rake:27:in `<main>'
Tasks: TOP => default => x86_64-darwin/libscrypt_ext.bundle
(See full trace by running task with --trace)

rake failed, exit code 1

Gem files will remain installed in /Users/paddor/.gem/ruby/2.3.1/gems/scrypt-3.0.3 for inspection.
Results logged to /Users/paddor/.gem/ruby/2.3.1/extensions/x86_64-darwin-15/2.3.0-static/scrypt-3.0.3/gem_make.out
gem install scrypt -v '3.0.3'  2.20s user 0.50s system 44% cpu 6.026 total
```

Any ideas?

**Comment by johbo** [comment]
Got the same issue. Running the command `clang -bundle ...` without the `-arch i386` works.

**Comment by jriverac** [comment]
@johbo Where did you run that? I have the same problem

**Comment by wtfiwtz** [comment]
Remove the references to `-arch i386` from `/Users/<username>/.rvm/gems/ruby-2.3.1/gems/scrypt-2.0.2/ext/scrypt/Rakefile` and `/Users/<username>/.rvm/gems/ruby-2.3.1/gems/scrypt-2.0.2/Rakefile` and then do this:

```
cd /Users/<username>/.rvm/gems/ruby-2.3.1/gems/scrypt-2.0.2/ext/scrypt
rake x86_64-darwin/libscrypt_ext.bundle
cd ../..
gem build scrypt.gemspec
gem install ./scrypt-2.0.2.gem
```

I also had to comment out a line about the private / secret key in `scrypt.gemspec`

**Comment by ddrscott** [comment]
I had an old version of llvm installed via homebrew.

`brew uninstall llvm` worked for me.

**Comment by lloeki** [comment]
> Remove the references to -arch i386

There's a PR for that: #71

**Comment by johanns** [comment]
Fix (PR #72) has been merged, and we should have official gem soon. Meanwhile you can build the gem from source (clone, and run `rake gem && gem install pkg/scrypt-3.0.6.gem` from source root), or grab unsigned gem file from here:  https://github.com/pbhogan/scrypt/releases/tag/3.0.6
